### DISCUSSION TITLE
ArgoCD sync status is not shown correctly on parent level.

### DISCUSSION THREAD

**Discussion by baht013** [argoproj/argo-cd#18230] [General]
Hello, 

I am using ArgoCD app of app pattern. Where i have root application and form there multiple child applications. I configured health via lue that works as expected. Right now i am having problem with sync status.

For example i have a change one of the child applications. When i open this child application i can see sync status is out of sync. 
When i open my parent application and look at the same child application there sync status is synced. 

Right now i am trying to figure out how to make this work properly, because i have some sync wave logic which i need.. 
Right now it doesn't work. Also worth to note i want manual sync. 

Any idea what i am missing?

**Comment by ChristianCiach** [comment]
I don't think this is supposed to work like that. When you are viewing your "root" app, then it is correct to display the child app as "synced", because this "child app" is just a kubernetes resource of the root app. Since the `kind: Application` manifest (just the manifest, not the resources of the the child app) is in sync, it correctly displays as synced. The root application doesn't know about the sync-status of the resources of the child app.

But I am curious why you are having issues with the sync wave logic. When you sync your root app to deploy the child apps, the sync-waves of the root app should wait for your child apps to become "healthy" (not "synced"). I can confirm that this is working if you've configured the health-checks correctly:


```yaml
  resource.customizations: |
    argoproj.io/Application:
      health.lua: |
        hs = {}
        hs.status = "Progressing"
        hs.message = ""
        if obj.status ~= nil then
          if obj.status.health ~= nil then
            hs.status = obj.status.health.status
            if obj.status.health.message ~= nil then
              hs.message = obj.status.health.message
            end
          end
        end
        return hs
```

**Comment by mattesja** [comment]
The provided `health.lua` script synchronized the `health` status between child and parent applications.
Therefore I wonder, if there's somehing similar for the `sync` status.

For us, it would be also helpful, to see the sync status of the child applications at the parent application:
- Better overview of the overall state
- Sync should be triggered from parent application

**Comment by baht013** [comment]
What i have found for Application kind sync wave works only for initial app deployment and deletion.. When comes updating child apps sync wave doesn't work anymore.

**Comment by CharlieR-o-o-t** [comment]
The same for me, argoCD shows "OutOfSync" status for child applications. But child app not synced at all and still sits on old TargetRevision. I met this case few times after update to v3.1.5+cfeed49.

Sometimes, application-controller restart helps.
